platform/security/idp/security-idp-server/src/test/groovy/org/codice/ddf/security/idp/server/IdpEndpointSpec.groovy

```diff
@@ -0,0 +1,475 @@
+                    getName() >> Idp.COOKIE
+                    getValue() >> cookieVal
+                }]
+            }
+        }
+
+        when:
+        def response = idpEndpoint.processRedirectLogout(null,
+                samlResponse,
+                relayState,
+                sigAlg,
+                sig,
+                request)
+
+        then:
+        notThrown(Exception)
+        idpEndpoint.logoutStates.decode(cookieVal).partialLogout
+    }
+
+    def "process post logout initial with only self sp"() {
+        setup:
+        def relayState = "relayState"
+        def samlRequest = RestSecurity.deflateAndBase64Encode("b64deflatedsamlstring")
+
+        def cookieVal = "cookievalue"
+        def nameId = "nameid"
+        def requestId = "originalRequestId"
+
+        idpEndpoint.logoutService = Mock(LogoutService) {
+            extractSamlLogoutRequest(_ as String) >> Mock(LogoutRequest) {
+                getIssuer() >> Mock(Issuer) {
+                    getValue() >> sp1entityid
+                }
+                getNameID() >> Mock(NameID) {
+                    getValue() >> nameId
+                }
+                getID() >> requestId
+            }
+            buildLogoutResponse(_ as String, _ as String, _ as String) >>
+                    {String issuer, String statusCode, String inResponseTo ->
+                        return new LogoutServiceImpl().buildLogoutResponse(issuer,
+                                statusCode,
+                                inResponseTo)
+                    }
+        }
+
+        def request = Mock(HttpServletRequest) {
+            getRequestURL() >> new StringBuffer("destinationurl")
+            getCookies() >> {
+                [Mock(Cookie) {
+                    getName() >> Idp.COOKIE
+                    getValue() >> cookieVal
+                }]
+            }
+        }
+
+        when:
+        def response = idpEndpoint.processPostLogout(samlRequest,
+                null,
+                relayState,
+                request)
+
+        then:
+        notThrown(Exception)
+        ((response.entity as String).contains("<form"))
+    }
+
+    def "xmlStreamException thrown from logout service"() {
+        setup:
+        def samlObject = RestSecurity.deflateAndBase64Encode("b64deflatedsamlstring")
+        idpEndpoint.logoutService = Mock(LogoutService) {
+            extractSamlLogoutRequest(_ as String) >> {throw new XMLStreamException()}
+            extractSamlLogoutResponse(_ as String) >> {throw new XMLStreamException()}
+        }
+        def request = Mock(HttpServletRequest) {
+            getRequestURL() >> new StringBuffer("destinationurl")
+
+            getCookies() >> {
+                [Mock(Cookie) {
+                    getName() >> Idp.COOKIE
+                    getValue() >> "bla"
+                }]
+            }
+        }
+
+        when:
+        idpEndpoint.processRedirectLogout(samlObject, null, "bla", "bla", "bla", request)
```

Comment:
Still not a fan of 'bla' repeatedly used as sample data, would be great if it were a bit more descriptive so it is at least some what readable.